### ISSUE TITLE
Not able to debug Share Extension in IOS

### ISSUE THREAD

**Issue by huyvuskedulo** [xamarin/xamarin-macios#19484]
I'm not able to debug share extension with current Xamarin.iOS

The error is returning as:

```
2023-11-18 10:49:03.985 mlaunch[9817:4884252] -[DTXMessage _appendTypesAndValues:withKey:list:]: unknown parameter type (13504896)
error HE0047: Failed to request XPC observation: Unable to invoke -[<DTXPCControlService: 0x6000018b1100> observeServicesWithIdentifier:requestingProcess:environment:arguments:options:] - too few arguments (4 provided, 5 expected) (1). Please try again.
        
```

### Steps to Reproduce

1. Create an iOS Project with Share extension
2. Build and deploy iOS Project
3. Try to connect and debug share extension

### Expected Behavior
Should be able to debug

### Actual Behavior
Not able to debug, returning as:
```
2023-11-18 10:49:03.985 mlaunch[9817:4884252] -[DTXMessage _appendTypesAndValues:withKey:list:]: unknown parameter type (13504896)
error HE0047: Failed to request XPC observation: Unable to invoke -[<DTXPCControlService: 0x6000018b1100> observeServicesWithIdentifier:requestingProcess:environment:arguments:options:] - too few arguments (4 provided, 5 expected) (1). Please try again.
        
```

### Environment

<details>
Visual Studio Community 2022 for Mac
Version 17.6.7 (build 417)
Installation UUID: ed3ff388-366a-49c7-ad26-f996be6930ec

Runtime
.NET 7.0.3 (64-bit)
Architecture: Arm64
Microsoft.macOS.Sdk 13.1.1007; git-rev-head:8afca776a0a96613dfb7200e0917bb57f9ed5583; git-branch:release/7.0.1xx-xcode14.2

Roslyn (Language Service)
4.6.0-3.23180.6+99e956e42697a6dd886d1e12478ea2b27cceacfa

NuGet
Version: 6.4.0.117

.NET SDK (Arm64)
SDK: /usr/local/share/dotnet/sdk/7.0.311/Sdks
SDK Versions:
	7.0.311
	7.0.310
	7.0.308
	7.0.304
	7.0.302
	7.0.203
	7.0.105
	7.0.100
	7.0.100-rc.1.22414.7
	6.0.417
	6.0.416
	6.0.414
	6.0.410
	6.0.408
	6.0.406
	6.0.405
	6.0.404
	6.0.403
	6.0.401
MSBuild SDKs: /Applications/Visual Studio.app/Contents/MonoBundle/MSBuild/Current/bin/Sdks

.NET SDK (x64)
SDK Versions:
	7.0.311
	7.0.310
	7.0.308
	7.0.304
	7.0.302
	7.0.203
	6.0.417
	6.0.416
	6.0.414
	6.0.410
	6.0.408

.NET Runtime (Arm64)
Runtime: /usr/local/share/dotnet/dotnet
Runtime Versions:
	7.0.14
	7.0.13
	7.0.11
	7.0.7
	7.0.5
	7.0.0
	7.0.0-rc.1.22411.12
	6.0.25
	6.0.24
	6.0.22
	6.0.18
	6.0.16
	6.0.14
	6.0.13
	6.0.12
	6.0.11
	6.0.9
	6.0.8

.NET Runtime (x64)
Runtime: /usr/local/share/dotnet/x64/dotnet
Runtime Versions:
	7.0.14
	7.0.13
	7.0.11
	7.0.7
	7.0.5
	6.0.25
	6.0.24
	6.0.22
	6.0.18
	6.0.16
	6.0.6
	6.0.5
	6.0.4
	5.0.17
	5.0.16
	3.1.29
	3.1.26
	3.1.25
	3.1.24

Xamarin.Profiler
Version: 1.8.0.49
Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler

Updater
Version: 11

Apple Developer Tools
Xcode: 14.3.1 21815
Build: 14E300c

Xamarin.Mac
Version: 9.3.0.23 Visual Studio Community
Hash: 9defd91b3
Branch: xcode14.3
Build date: 2023-10-23 16:14:59-0400

Xamarin.iOS
Version: 16.4.0.23 Visual Studio Community
Hash: 9defd91b3
Branch: xcode14.3
Build date: 2023-10-23 16:15:00-0400

Xamarin Designer
Version: 17.6.3.9
Hash: 2648399ae8
Branch: remotes/origin/d17-6
Build date: 2023-11-10 12:19:46 UTC

Xamarin.Android
Version: 13.2.2.0 (Visual Studio Community)
Commit: xamarin-android/d17-5/45b0e14
Android SDK: /Users/huyvu/Library/Android/sdk
	Supported Android versions:
		12.1 (API level 32)
		12.0 (API level 31)
		11.0 (API level 30)
		10.0 (API level 29)
		13.0 (API level 33)

SDK Command-line Tools Version: 7.0
SDK Platform Tools Version: 34.0.1
SDK Build Tools Version: 34.0.0 rc3

Build Information: 
Mono: d9a6e87
Java.Interop: xamarin/java.interop/d17-5@149d70fe
SQLite: xamarin/sqlite/3.40.1@68c69d8
Xamarin.Android Tools: xamarin/xamarin-android-tools/d17-5@ca1552d

Microsoft Build of OpenJDK
Java SDK: /Library/Java/JavaVirtualMachines/microsoft-11.jdk
11.0.16.1
Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

Eclipse Temurin JDK
Java SDK: /Library/Java/JavaVirtualMachines/temurin-8.jdk
1.8.0.302
Android Designer EPL code available here:
https://github.com/xamarin/AndroidDesigner.EPL

Android SDK Manager
Version: 17.6.0.50
Hash: a715dca
Branch: HEAD
Build date: 2023-11-10 12:19:52 UTC

Android Device Manager
Version: 0.0.0.1309
Hash: 06e3e77
Branch: HEAD
Build date: 2023-11-10 12:19:52 UTC

Build Information
Release ID: 1706070417
Git revision: 179aa48a5702754e41a8218d0403ddc3be46706f
Build date: 2023-11-10 12:18:09+00
Build branch: release-17.6
Build lane: release-17.6

Operating System
Mac OS X 14.1.1
Darwin 23.1.0 Darwin Kernel Version 23.1.0
    Mon Oct  9 21:27:24 PDT 2023
    root:xnu-10002.41.9~6/RELEASE_ARM64_T6000 arm64

</details>

**Comment by rolfbjarne** [comment]
Yes, this is because of iOS 17 / Xcode 15. Apple completely rewrote their libraries to interact with devices, and we'll have to reimplement our side of it.
